gradient/commands/datasets.py

```diff
@@ -599,8 +692,13 @@ def _sign_and_put(self, dataset_version_id, pool, results, update_status):
 
         for pre_signed, result in zip(pre_signeds, results):
             update_status()
-            pool.put(self._put, url=pre_signed.url,
-                     path=result['path'], content_type=result['mimetype'])
+            pool.put(
```

Comment:
Granted, this isn't really ideal. We're single-threading all parts of an upload in a single worker rather than distributing all N parts among all M workers in the pool. This will result in longer upload times, but that's better than the broken upload we have today. Soooo baby steps.